Consider the update function below from counterClient.js and answer the questions below.

function update(){
	$.ajax({ 
		method: "PUT", 
		url: "/api/counter/"+$("#updateCounterName").val()+"/", 
		data: JSON.stringify({ "amount": $("#updateCounterAmount").val() }),
		processData: false, 
		contentType: "application/json; charset=utf-8",
		dataType :"json"
	}).done(function(data, text_status, jqXHR){
		console.log(jqXHR.status+" "+text_status+JSON.stringify(data));

		retrieveAll();
	}).fail(function(err){
		console.log("fail "+err.status+" "+JSON.stringify(err.responseJSON));
	});
}


q1: Explain each of the lines in the ajax call
(a) method: "PUT"

(b) url: "/api/counter/"+$("#updateCounterName").val()+"/",

(c) data: JSON.stringify({ "amount": $("#updateCounterAmount").val() }),

(d) processData: false,

(e) contentType: "application/json; charset=utf-8",

(f) dataType :"json"

q2: When does the .done callback get called? Hint: Play with the application

q3: When function(data, text_status, jqXHR) is called, what will be in 
(a) data
(b) text_status
(c) jqHXR.status

q4: When does the .fail callback get called? 

q5: When function(err) is called, what will be in 
(a) err.status
(b) err.responseJSON

